wren-ai-service/src/providers/llm/azure_openai.py

```diff
@@ -125,228 +112,38 @@ async def run(
-            or documents
-            and not isinstance(documents[0], Document)
-        ):
-            raise TypeError(
-                "AzureOpenAIDocumentEmbedder expects a list of Documents as input."
-                "In case you want to embed a string, please use the AzureOpenAITextEmbedder."
-            )
-
-        logger.info(
-            f"Running Async OpenAI document embedder with documents: {documents}"
-        )
-
-        texts_to_embed = self._prepare_texts_to_embed(documents=documents)
-
-        embeddings, meta = await self._embed_batch(
-            texts_to_embed=texts_to_embed, batch_size=self.batch_size
-        )
-
-        for doc, emb in zip(documents, embeddings):
-            doc.embedding = emb
-
-        return {"documents": documents, "meta": meta}
-
-
-@provider("azure_openai")
+@provider("azure_openai_llm")
 class AzureOpenAILLMProvider(LLMProvider):
     def __init__(
         self,
-        chat_api_key: Secret = Secret.from_env_var("AZURE_CHAT_KEY"),
-        chat_api_base: str = os.getenv("AZURE_CHAT_BASE"),
-        embed_api_key: Secret = Secret.from_env_var("AZURE_EMBED_KEY"),
-        embed_api_base: str = os.getenv("AZURE_EMBED_BASE"),
-        chat_api_version: str = os.getenv("AZURE_CHAT_VERSION"),
-        embed_api_version: str = os.getenv("AZURE_EMBED_VERSION"),
-        generation_model: str = os.getenv("AZURE_GENERATION_MODEL")
-        or AZURE_GENERATION_MODEL,
-        embedding_model: str = os.getenv("EMBEDDING_MODEL") or EMBEDDING_MODEL_NAME,
-        embedding_model_dim: int = (
-            int(os.getenv("EMBEDDING_MODEL_DIMENSION"))
-            if os.getenv("EMBEDDING_MODEL_DIMENSION")
-            else 0
-        )
-        or EMBEDDING_MODEL_DIMENSION,
+        chat_api_key: Secret = Secret.from_env_var("LLM_AZURE_OPENAI_API_KEY"),
+        chat_api_base: str = os.getenv("LLM_AZURE_OPENAI_API_BASE"),
+        chat_api_version: str = os.getenv("LLM_AZURE_OPENAI_VERSION"),
+        generation_model: str = os.getenv("GENERATION_MODEL") or GENERATION_MODEL,
     ):
-        logger.info(f"Using Azure OpenAI Generation Model: {generation_model}")
+        logger.info(f"Using Azure OpenAI LLM: {generation_model}")
 
         self._generation_api_key = chat_api_key
         self._generation_api_base = chat_api_base
         self._generation_api_version = chat_api_version
         self._generation_model = generation_model
-        self._embedding_api_base = embed_api_base
-        self._embedding_api_key = embed_api_key
-        self._embedding_api_version = embed_api_version
-        self._embedding_model = embedding_model
-        self._embedding_model_dim = embedding_model_dim
 
     def get_generator(
         self,
-        model_kwargs: Optional[Dict[str, Any]] = AZURE_GENERATION_MODEL_KWARGS,
+        model_kwargs: Dict[str, Any] = orjson.loads(
+            os.getenv("GENERATION_MODEL_KWARGS", "{}")
+        )
+        or GENERATION_MODEL_KWARGS,
```

Comment:
it is a gentle comment to mention here has the same situation that we mentioned in other comment! 😆